swift/src/Ice/CommunicatorI.swift

```diff
@@ -296,11 +297,27 @@ extension Communicator {
 }
 
 struct DefaultsAndOverrides {
-    init(handle: ICECommunicator) {
-        var defaultEncoding = EncodingVersion()
-        handle.getDefaultEncoding(major: &defaultEncoding.major, minor: &defaultEncoding.minor)
-        self.defaultEncoding = defaultEncoding
-        defaultFormat = FormatType(rawValue: handle.getDefaultFormat())!
+    init(_ properties: Properties) throws {
+        let defaultEncodingValue = properties.getIceProperty("Ice.Default.EncodingVersion")
+        switch defaultEncodingValue {
+        case "1.0":
+            defaultEncoding = EncodingVersion(major: 1, minor: 0)
+        case "1.1":
+            defaultEncoding = EncodingVersion(major: 1, minor: 1)
+        default:
+            fatalError("Invalid Ice.Default.EncodingVersion value: \(defaultEncodingValue)")
+        }
+
+        let defaultFormatValue = try properties.getIcePropertyAsInt("Ice.Default.SlicedFormat")
+
+        switch defaultFormatValue {
+        case 0:
+            defaultFormat = .compactFormat
+        case 1:
```

Comment:
According to https://doc.zeroc.com/ice/3.7/property-reference/ice-default#id-.Ice.Default.*v3.7-Ice.Default.SlicedFormat

0 = compact
non-0 = sliced

That's not what you implemented.